### DISCUSSION TITLE
Twitter ending free access to the Twitter API

### DISCUSSION THREAD

**Discussion by codeXbkX** [plhery/node-twitter-api-v2#448] [General]
End of twitter bot era ?

> Starting February 9, we will no longer support free access to the Twitter API, both v2 and v1.1. A paid basic tier will be available instead 

[Official Tweet](https://twitter.com/TwitterDev/status/1621026986784337922)

**Comment by Zedifuu** [comment]
Honestly quite a disappointing move from Twitter(Ultimately Elon). First it went with tweet source data being removed, but my bot continued to function (Just a wee crossposter for Discord...) Now Twitter are taking a more aggressive stance moving to this paid model... Not to mention only giving a weeks notice...

I really do hope they have reasonable model(which I doubt), available on the Dev Portal.. and don't completely refactor their endpoints.

It is a really sorry state of affairs, one of the replies on that Twitter post resonated with me: 
> twitter really is in a competition with netflix to see who can ruin their valuation the fastest, huh

**Comment by plhery** [comment]
Yes it's sad, so many developers worked on improving the Twitter experience, some became businesses, for nothing and with a short notice period.
This new Twitter does not respect its employees users developers and ecosystem.

Alternative hacky possibilities left (I may advertise in the readme?):
- Using the official client tokens which are actual API tokens with less limitations
- Building a Twitter web scrapping library

**Comment by EdwardChristian** [comment]
My university final year project is reliant on the twitter api being free, spent 5-6 months developing it and now I have 6 weeks to completely revamp the frontend and backend to use a different social platform... otherwise I fail my degree.

Completely screwed me right at the end..
